manifests/init.pp

```diff
@@ -155,6 +158,7 @@
   $user_hash          = {},
   $configure_firewall = undef,
   $install_java       = $jenkins::params::install_java,
+  $java_package_name  = undef,
```

Comment:
Yiou aren't specifying any default.  Is a default needed?  If yes, then we should probably be dealing with it in params.pp, and pulling the setting in here., similar to the prior line.